corehq/apps/data_interfaces/views.py

```diff
@@ -651,20 +651,18 @@ class AutomaticUpdateRuleListView(DataInterfaceSection):
     def dispatch(self, request, *args, **kwargs):
         return super().dispatch(request, *args, **kwargs)
 
-    def get(self, request, *args, **kwargs):
+    @property
+    def page_context(self):
         context = super().main_context
```

Comment:
Didn't notice this earlier: is it intentional this starts with super's main_context instead of super's page_context? In general you don't need to think about main_context, it just pulls from the `page_title`, etc attributes and ultimately gets [combined with page_context](https://github.com/dimagi/commcare-hq/blob/14793c049a2a2691a02db5660d015166b0b1913e/corehq/apps/hqwebapp/views.py#L902-L906).

I'm being picky because our views are inconsistent about how they handle the page_context/main_context and the navigation attributes, which makes our navigation annoying inconsistent. Reminds me of these improvements Charl just made: https://github.com/dimagi/commcare-hq/pull/32179

Reply:
Again just based on the view I was using as a reference. That makes a lot of sense and thanks for being picky! The inconsistencies only add to the confusion so it is well worth resolving. See 56b14a5513b58cf87db5a21d1ff72e7767f0312a